--- busybox-1.17.3/networking/udhcp/common.c~	2010-10-09 22:57:14.000000000 +0300
+++ busybox-1.17.3/networking/udhcp/common.c	2010-11-08 21:36:56.660518687 +0200
@@ -32,8 +32,8 @@
 	{ OPTION_STRING               | OPTION_REQ, 0x0c }, /* DHCP_HOST_NAME     */
 	{ OPTION_U16                              , 0x0d }, /* DHCP_BOOT_SIZE     */
 	{ OPTION_STRING               | OPTION_REQ, 0x0f }, /* DHCP_DOMAIN_NAME   */
-	{ OPTION_IP                               , 0x10 }, /* DHCP_SWAP_SERVER   */
-	{ OPTION_STRING                           , 0x11 }, /* DHCP_ROOT_PATH     */
+	{ OPTION_IP                   | OPTION_REQ, 0x10 }, /* DHCP_SWAP_SERVER   */
+	{ OPTION_STRING               | OPTION_REQ, 0x11 }, /* DHCP_ROOT_PATH     */
 	{ OPTION_U8                               , 0x17 }, /* DHCP_IP_TTL        */
 	{ OPTION_U16                              , 0x1a }, /* DHCP_MTU           */
 	{ OPTION_IP                   | OPTION_REQ, 0x1c }, /* DHCP_BROADCAST     */
