diff -urN busybox-1.00-pre10.org/networking/udhcp/options.c busybox-1.00-pre10/networking/udhcp/options.c
--- busybox-1.00-pre10.org/networking/udhcp/options.c	2004-06-05 20:24:43.698887752 +0200
+++ busybox-1.00-pre10/networking/udhcp/options.c	2004-06-05 20:26:15.241971088 +0200
@@ -27,8 +27,8 @@
 	{"hostname",	OPTION_STRING | OPTION_REQ,		0x0c},
 	{"bootsize",	OPTION_U16,				0x0d},
 	{"domain",	OPTION_STRING | OPTION_REQ,		0x0f},
-	{"swapsvr",	OPTION_IP,				0x10},
-	{"rootpath",	OPTION_STRING,				0x11},
+	{"swapsvr",	OPTION_IP | OPTION_REQ,			0x10},
+	{"rootpath",	OPTION_STRING | OPTION_REQ,		0x11},
 	{"ipttl",	OPTION_U8,				0x17},
 	{"mtu",		OPTION_U16,				0x1a},
 	{"broadcast",	OPTION_IP | OPTION_REQ,			0x1c},
@@ -39,8 +39,8 @@
 	{"dhcptype",	OPTION_U8,				0x35},
 	{"serverid",	OPTION_IP,				0x36},
 	{"message",	OPTION_STRING,				0x38},
-	{"tftp",	OPTION_STRING,				0x42},
-	{"bootfile",	OPTION_STRING,				0x43},
+	{"tftp",	OPTION_STRING | OPTION_REQ,		0x42},
+	{"bootfile",	OPTION_STRING | OPTION_REQ,		0x43},
 	{"",		0x00,				0x00}
 };
 
