diff -Nur --exclude *~ busybox-0.51/busybox.c busybox-0.51.new/busybox.c
--- busybox-0.51/busybox.c	Tue Apr 10 00:48:11 2001
+++ busybox-0.51.new/busybox.c	Sun May 20 12:43:53 2001
@@ -94,6 +94,8 @@
 	}
 
 #ifdef BB_SH
+	if (strcmp(applet_name, "lash")==0) applet_name="sh";
+
 	/* Add in a special case hack -- whenever **argv == '-'
 	 * (i.e. '-su' or '-sh') always invoke the shell */
 	if (**argv == '-' && *(*argv+1)!= '-') {
