diff -urN busybox-1.00-pre2.org/modutils/insmod.c busybox-1.00-pre2/modutils/insmod.c
--- busybox-1.00-pre2.org/modutils/insmod.c	2003-08-02 23:46:49.000000000 +0200
+++ busybox-1.00-pre2/modutils/insmod.c	2003-08-05 23:08:15.000000000 +0200
@@ -1646,6 +1646,9 @@
 	int n_binding = ELFW(ST_BIND) (info);
 
 	for (sym = f->symtab[hash]; sym; sym = sym->next)
+		if (strncmp((char *)sym->name, "GPLONLY_", 8) == 0)
+			((char *)sym->name) += 8;
+	
 		if (f->symbol_cmp(sym->name, name) == 0) {
 			int o_secidx = sym->secidx;
 			int o_info = sym->info;

