--- busybox-1.00/networking/ftpgetput.c.orig	2005-02-15 13:30:36.000000000 +0100
+++ busybox-1.00/networking/ftpgetput.c	2005-02-15 14:00:35.315575696 +0100
@@ -237,10 +237,6 @@
 	}
 	fd_data = xconnect_ftpdata(server, buf);
 
-	if (ftpcmd("CWD ", server_path, control_stream, buf) != 250) {
-		bb_error_msg_and_die("CWD error: %s", buf + 4);
-	}
-
 	/* get the local file */
 	if ((local_path[0] == '-') && (local_path[1] == '\0')) {
 		fd_local = STDIN_FILENO;
@@ -264,7 +260,7 @@
 			break;
 		}
 	}
-	response = ftpcmd("STOR ", local_path, control_stream, buf);
+	response = ftpcmd("STOR ", server_path, control_stream, buf);
 	switch (response) {
 	case 125:
 	case 150:
