diff -durN -x '*~' -x '*.o' -x '.*' -x '*.[0-9]' -x '*.links' -x '*.html' -x '*.txt' -x '*.pod' -x 'config*' busybox-1.00-pre4.orig/libbb/loop.h busybox-1.00-pre4/libbb/loop.h
--- busybox-1.00-pre4.orig/libbb/loop.h	1970-01-01 01:00:00.000000000 +0100
+++ busybox-1.00-pre4/libbb/loop.h	2004-12-21 17:45:14.000000000 +0100
@@ -0,0 +1,5 @@
+#include <linux/posix_types.h>
+#undef dev_t
+#define dev_t __kernel_dev_t
+#include <linux/loop.h>
+#undef dev_t
diff -durN -x '*~' -x '*.o' -x '.*' -x '*.[0-9]' -x '*.links' -x '*.html' -x '*.txt' -x '*.pod' -x 'config*' busybox-1.00-pre4.orig/miscutils/hdparm.c busybox-1.00-pre4/miscutils/hdparm.c
--- busybox-1.00-pre4.orig/miscutils/hdparm.c	2004-12-21 17:58:28.111674208 +0100
+++ busybox-1.00-pre4/miscutils/hdparm.c	2004-12-21 17:53:25.000000000 +0100
@@ -45,7 +45,8 @@
 #include <linux/types.h>
 #include <linux/hdreg.h>
 #include <linux/major.h>
-#include <asm/byteorder.h>
+#include <endian.h>
+#include <byteswap.h>
 
 
 #if (__BYTE_ORDER == __BIG_ENDIAN) && !defined(__USE_XOPEN)
@@ -2425,7 +2425,8 @@
 			}
 		}
 		for(i=0; i<(sizeof args)/2; i+=2)
-		    __le16_to_cpus((uint16_t *)(&args[i]));
+			if (__BYTE_ORDER != __LITTLE_ENDIAN)
+				args[i]=bswap_16(args[i]);
 
 		identify((void *)&args[4], NULL);
 identify_abort:
@@ -2491,7 +2492,8 @@
 	for (i = 0; count >= 4; ++i)
 	{
 		sbuf[i] = (fromhex(b[0]) << 12) | (fromhex(b[1]) << 8) | (fromhex(b[2]) << 4) | fromhex(b[3]);
-		__le16_to_cpus((uint16_t *)(&sbuf[i]));
+		if (__BYTE_ORDER != __LITTLE_ENDIAN)
+			sbuf[i]=bswap_16(sbuf[i]);
 		b += 5;
 		count -= 5;
 	}
