diff -urN busybox-1.00-pre2/include/libbb.h busybox-1.00-pre2.org/include/libbb.h
--- busybox-1.00-pre2/include/libbb.h	2003-08-02 23:40:57.000000000 +0200
+++ busybox-1.00-pre2.org/include/libbb.h	2003-08-02 23:45:09.000000000 +0200
@@ -346,7 +346,7 @@
 # define VC_2 "/dev/vc/2"
 # define VC_3 "/dev/vc/3"
 # define VC_4 "/dev/vc/4"
-# define VC_5 "/dev/vc/5"
+# define VC_12 "/dev/vc/12"
 #if defined(__sh__) || defined(__H8300H__) || defined(__H8300S__)
 /* Yes, this sucks, but both SH (including sh64) and H8 have a SCI(F) for their
    respective serial ports .. as such, we can't use the common device paths for
@@ -367,7 +367,7 @@
 # define VC_2 "/dev/tty2"
 # define VC_3 "/dev/tty3"
 # define VC_4 "/dev/tty4"
-# define VC_5 "/dev/tty5"
+# define VC_12 "/dev/tty12"
 #if defined(__sh__) || defined(__H8300H__) || defined(__H8300S__)
 #  define SC_0 "/dev/ttySC0"
 #  define SC_1 "/dev/ttySC1"
diff -urN busybox-1.00-pre2/init/init.c busybox-1.00-pre2.org/init/init.c
--- busybox-1.00-pre2/init/init.c	2003-08-02 23:40:57.000000000 +0200
+++ busybox-1.00-pre2.org/init/init.c	2003-08-02 23:45:30.000000000 +0200
@@ -155,7 +155,7 @@
 static char console[CONSOLE_BUFF_SIZE] = _PATH_CONSOLE;
 
 #if !ENABLE_SYSLOGD
-static char *log_console = VC_5;
+static char *log_console = VC_12;
 #endif
 #if !ENABLE_DEBUG_INIT
 static sig_atomic_t got_cont = 0;
