--- busybox-1.3.1/networking/udhcp/options.c.orig	2006-12-27 05:52:33.000000000 +0100
+++ busybox-1.3.1/networking/udhcp/options.c	2006-12-30 12:31:57.305539547 +0100
@@ -24,8 +24,8 @@
 	{"hostname",    OPTION_STRING | OPTION_REQ,             0x0c},
 	{"bootsize",    OPTION_U16,                             0x0d},
 	{"domain",      OPTION_STRING | OPTION_LIST | OPTION_REQ, 0x0f},
-	{"swapsvr",     OPTION_IP,                              0x10},
-	{"rootpath",    OPTION_STRING,                          0x11},
+	{"swapsvr",     OPTION_IP | OPTION_REQ,                 0x10},
+	{"rootpath",    OPTION_STRING | OPTION_REQ,             0x11},
 	{"ipttl",       OPTION_U8,                              0x17},
 	{"mtu",         OPTION_U16,                             0x1a},
 	{"broadcast",   OPTION_IP | OPTION_REQ,                 0x1c},

