--- busybox-1.00-pre10/miscutils/hdparm.c_	2004-04-29 07:01:59.000000000 +0000
+++ busybox-1.00-pre10/miscutils/hdparm.c	2004-04-29 07:33:15.087216840 +0000
@@ -1227,7 +1227,7 @@
 static int flagcount = 0, do_flush = 0, is_scsi_hd = 0, is_xt_hd = 0;
 static int do_ctimings, do_timings = 0;
 
-static unsigned long set_readahead= 0, get_readahead= 0, readahead= 0;
+static unsigned long set_readahead= 0, get_readahead= 0, read_ahead= 0;
 static unsigned long set_readonly = 0, get_readonly = 0, readonly = 0;
 static unsigned long set_unmask   = 0, get_unmask   = 0, unmask   = 0;
 static unsigned long set_mult     = 0, get_mult     = 0, mult     = 0;
@@ -1937,8 +1937,8 @@
 
 	if (set_readahead)
 	{
-		if_printf(get_readahead," setting fs readahead to %ld\n", readahead);
-		bb_ioctl(fd, BLKRASET,(int *)readahead,"BLKRASET");
+		if_printf(get_readahead," setting fs readahead to %ld\n", read_ahead);
+		bb_ioctl(fd, BLKRASET,(int *)read_ahead,"BLKRASET");
 	}
 #ifdef CONFIG_FEATURE_HDPARM_HDIO_UNREGISTER_HWIF
 	if (unregister_hwif)
@@ -2832,7 +2832,7 @@
 						noisy = 1;
 						if (!*p && argc && isalnum(**argv))
 							p = *argv++, --argc;
-						p=GET_NUMBER(p,&set_readahead,&readahead);
+						p=GET_NUMBER(p,&set_readahead,&read_ahead);
 						break;
 					case 'B':
 						get_apmmode = noisy;
