diff -urN busybox-1.00-pre2.org/libbb/syscalls.c busybox-1.00-pre2/libbb/syscalls.c
--- busybox-1.00-pre2.org/libbb/syscalls.c	2003-08-02 23:31:11.000000000 +0200
+++ busybox-1.00-pre2/libbb/syscalls.c	2003-08-02 23:38:37.000000000 +0200
@@ -29,6 +29,21 @@
 #include <sys/syscall.h>
 #include "libbb.h"
 
+#ifndef __NR_pivot_root
+# warning "Guessing pivot_root() syscall number"
+# ifdef __i386__
+#  define __NR_pivot_root 217
+# elif defined(__alpha__)
+#  define __NR_pivot_root 374
+# elif defined(__sparc__) || defined(__sparc64__)
+#  define __NR_pivot_root 146
+# elif defined(__powerpc__)
+#  define __NR_pivot_root 203
+# else
+#  error "Get kernel-2.4 headers!"
+# endif
+#endif
+
 int sysfs( int option, unsigned int fs_index, char * buf)
 {
 	return(syscall(__NR_sysfs, option, fs_index, buf));
