--- busybox-1.2.0/modutils/insmod.c.orig	2006-07-01 00:42:04.000000000 +0200
+++ busybox-1.2.0/modutils/insmod.c	2006-07-20 19:42:23.188629500 +0200
@@ -249,7 +249,17 @@
 #endif
 
 /* S390 */
-#if defined(__s390__)
+#if defined(__s390x__)
+#define MATCH_MACHINE(x) (x == EM_S390)
+#define SHT_RELM	SHT_RELA
+#define Elf64_RelM	Elf64_Rela
+#define ELFCLASSM	ELFCLASS64
+#define CONFIG_USE_PLT_ENTRIES
+#define CONFIG_PLT_ENTRY_SIZE 8
+#define CONFIG_USE_GOT_ENTRIES
+#define CONFIG_GOT_ENTRY_SIZE 8
+#define CONFIG_USE_SINGLE
+#elif defined(__s390__)
 #define MATCH_MACHINE(x) (x == EM_S390)
 #define SHT_RELM	SHT_RELA
 #define Elf32_RelM	Elf32_Rela
@@ -283,10 +293,17 @@
 
 /* Sparc */
 #if defined(__sparc__)
+#  if defined(__arch64__)
+#define MATCH_MACHINE(x) ((x) == EM_SPARCV9 || (x) == EM_SPARC64)
+#define SHT_RELM       SHT_RELA
+#define Elf64_RelM     Elf64_Rela
+#define ELFCLASSM      ELFCLASS64
+#  else
 #define MATCH_MACHINE(x) (x == EM_SPARC)
 #define SHT_RELM       SHT_RELA
 #define Elf32_RelM     Elf32_Rela
 #define ELFCLASSM      ELFCLASS32
+#  endif
 #endif
 
 /* v850e */
