--- Config.h	2003-07-01 03:06:16.000000000 +0200
+++ Config.h	2003-07-01 03:07:21.000000000 +0200
@@ -9,14 +9,14 @@
 // BusyBox Applications
 //#define BB_ADJTIMEX
 //#define BB_AR
-#define BB_ASH
+//#define BB_ASH
 #define BB_BASENAME
 #define BB_CAT
 #define BB_CHGRP
 #define BB_CHMOD
 #define BB_CHOWN
 #define BB_CHROOT
-#define BB_CHVT
+//#define BB_CHVT
 #define BB_CLEAR
 //#define BB_CMP
 #define BB_CP
@@ -36,63 +36,62 @@
 #define BB_DU
 //#define BB_DUMPKMAP
 #define BB_ECHO
-#define BB_ENV
-//#define BB_EXPR
+//#define BB_ENV
+#define BB_EXPR
 //#define BB_FBSET
 //#define BB_FDFLUSH
 #define BB_FIND
 #define BB_FREE
-//#define BB_FREERAMDISK
+#define BB_FREERAMDISK
 //#define BB_FSCK_MINIX
 //#define BB_GETOPT
 #define BB_GREP
 #define BB_GUNZIP
 #define BB_GZIP
-#define BB_HALT
+//#define BB_HALT
 #define BB_HEAD
 //#define BB_HOSTID
-//#define BB_HOSTNAME
+#define BB_HOSTNAME
 //#define BB_HUSH
-#define BB_ID
-//#define BB_IFCONFIG
+//#define BB_ID
+#define BB_IFCONFIG
 #define BB_INIT
-//#define BB_INSMOD
+#define BB_INSMOD
 #define BB_INSMOD_NG
 #define BB_KILL
 #define BB_KILLALL
 #define BB_KLOGD
-//#define BB_LASH
+#define BB_LASH
 //#define BB_LENGTH
 #define BB_LN
-//#define BB_LOADACM
-//#define BB_LOADFONT
+#define BB_LOADACM
+#define BB_LOADFONT
 //#define BB_LOADKMAP
-#define BB_LOGGER
+//#define BB_LOGGER
 //#define BB_LOGNAME
-//#define BB_LOSETUP
 #define BB_LS
 #define BB_LSMOD
-//#define BB_MAKEDEVS
-//#define BB_MD5SUM
+#define BB_MAKEDEVS
+#define BB_MD5SUM
 #define BB_MKDIR
 //#define BB_MKFIFO
 //#define BB_MKFS_MINIX
 #define BB_MKNOD
 #define BB_MKSWAP
-//#define BB_MKTEMP
+#define BB_MKTEMP
 #define BB_MODPROBE
 #define BB_MORE
 #define BB_MOUNT
 //#define BB_MSH
 //#define BB_MT
 #define BB_MV
-//#define BB_NC
+#define BB_NC
 //#define BB_NSLOOKUP
-#define BB_PIDOF
-//#define BB_PING
-//#define BB_PIVOT_ROOT
+//#define BB_PIDOF
+#define BB_PING
+#define BB_PIVOT_ROOT
 #define BB_POWEROFF
-//#define BB_PRINTF
+#define BB_PRINTF
 #define BB_PS
 #define BB_PWD
 #define BB_RAID_START
@@ -103,7 +102,7 @@
 #define BB_RESET
 #define BB_RM
 #define BB_RMDIR
-//#define BB_RMMOD
+#define BB_RMMOD
 //#define BB_ROUTE
 //#define BB_RPM2CPIO
 #define BB_SED
@@ -116,34 +115,32 @@
 #define BB_SYSLOGD
 #define BB_TAIL
 #define BB_TAR
-//#define BB_TEE
-//#define BB_TEST
-//#define BB_TELNET
+#define BB_TEE
+#define BB_TEST
+#define BB_TELNET
 //#define BB_TFTP
-//#define BB_TIME
-//#define BB_TOP
 #define BB_TOUCH
-//#define BB_TR
+#define BB_TR
 //#define BB_TRACEROUTE
 #define BB_TRUE_FALSE
-#define BB_TTY
+//#define BB_TTY
 //#define BB_UNIX2DOS
 //#define BB_UUENCODE
 //#define BB_UUDECODE
 #define BB_UMOUNT
 #define BB_UNIQ
 #define BB_UNAME
-//#define BB_UPDATE
-#define BB_UPTIME
-//#define BB_USLEEP
+#define BB_UPDATE
+//#define BB_UPTIME
+#define BB_USLEEP
 //#define BB_VI
 //#define BB_WATCHDOG
-#define BB_WC
+//#define BB_WC
 //#define BB_WGET
 #define BB_WHICH
-#define BB_WHOAMI
+//#define BB_WHOAMI
 #define BB_XARGS
-#define BB_YES
+//#define BB_YES
 // End of Applications List
 //
 //
@@ -184,16 +181,16 @@
 //#define BB_FEATURE_USE_DEVPS_PATCH
 //
 // show verbose usage messages
-#define BB_FEATURE_VERBOSE_USAGE
+//#define BB_FEATURE_VERBOSE_USAGE
 //
 // Use termios to manipulate the screen ('more' is prettier with this on)
-//#define BB_FEATURE_USE_TERMIOS
+#define BB_FEATURE_USE_TERMIOS
 //
 // calculate terminal & column widths (for more, ls, and telnet)
 #define BB_FEATURE_AUTOWIDTH
 //
 // show username/groupnames for ls
-#define BB_FEATURE_LS_USERNAME
+//#define BB_FEATURE_LS_USERNAME
 //
 // show file timestamps in ls
 #define BB_FEATURE_LS_TIMESTAMPS
@@ -208,25 +205,25 @@
 #define BB_FEATURE_LS_RECURSIVE
 //
 // enable ls -L
-#define BB_FEATURE_LS_FOLLOWLINKS
+//#define BB_FEATURE_LS_FOLLOWLINKS
 //
 // Use color to identify different file types
-#define BB_FEATURE_LS_COLOR
+//#define BB_FEATURE_LS_COLOR
 //
 // Disable for a smaller (but less functional) ping
-#define BB_FEATURE_FANCY_PING
+//#define BB_FEATURE_FANCY_PING
 //
 // Make init use a simplified /etc/inittab file (recommended).
 #define BB_FEATURE_USE_INITTAB
 //
 //Enable init being called as /linuxrc
-#define BB_FEATURE_LINUXRC
+//#define BB_FEATURE_LINUXRC
 //
 //Have init enable core dumping for child processes (for debugging only) 
 //#define BB_FEATURE_INIT_COREDUMPS
 //
 //Make sure nothing is printed to the console on boot
-//#define BB_FEATURE_EXTRA_QUIET
+#define BB_FEATURE_EXTRA_QUIET
 //
 // enable syslogd -R remotehost
 #define BB_FEATURE_REMOTE_LOG
@@ -236,10 +233,10 @@
 //
 //Disable for a simple tail implementation (2.34k vs 3k for the full one).
 //Both provide 'tail -f', but this cuts out -c, -q, -s, and -v. 
-#define BB_FEATURE_FANCY_TAIL
+//#define BB_FEATURE_FANCY_TAIL
 //
 // Enable support for loop devices in mount
-#define BB_FEATURE_MOUNT_LOOP
+//#define BB_FEATURE_MOUNT_LOOP
 //
 // Enable support for a real /etc/mtab file instead of /proc/mounts
 //#define BB_FEATURE_MTAB_SUPPORT
@@ -250,7 +247,7 @@
 //
 // If you are using uClibc, be sure that you've already compiled
 // uClibc with INCLUDE_RPC=true (contained in the Config file)
-//#define BB_FEATURE_NFSMOUNT
+#define BB_FEATURE_NFSMOUNT
 //
 // Enable support forced filesystem unmounting 
 // (i.e., in case of an unreachable NFS system).
@@ -263,13 +260,13 @@
 #define BB_FEATURE_TAR_EXCLUDE
 //
 // Enable support for tar -z option (currently only works for inflating)
-#define BB_FEATURE_TAR_GZIP 
+//#define BB_FEATURE_TAR_GZIP 
 //
 // Enable reverse sort
 #define BB_FEATURE_SORT_REVERSE
 //
 // Enable uniqe sort
-#define BB_FEATURE_SORT_UNIQUE
+//#define BB_FEATURE_SORT_UNIQUE
 //
 // Enable command line editing in the shell.  
 // Only relevant if a shell is enabled. On by default.
@@ -286,7 +283,7 @@
 //were shell builtins.  Nice for staticly linking an emergency rescue shell,
 //among other things. Off by default.
 // Only relevant if a shell is enabled.
-//#define BB_FEATURE_SH_STANDALONE_SHELL
+#define BB_FEATURE_SH_STANDALONE_SHELL
 //
 //When this is enabled, busybox shell applets can be called using full path
 //names.  This causes applets (i.e., most busybox commands) to override
@@ -367,7 +364,7 @@
 //#define BB_FEATURE_CLEAN_UP
 //
 // Support for human readable output by ls, du, etc.(example 13k, 23M, 235G)
-#define BB_FEATURE_HUMAN_READABLE
+//#define BB_FEATURE_HUMAN_READABLE
 //
 // Support for the find -type option.
 #define BB_FEATURE_FIND_TYPE
