--- busybox-1.12.2/console-tools/loadfont.c~	2008-11-19 12:27:34.314700966 +0100
+++ busybox-1.12.2/console-tools/loadfont.c	2008-11-19 12:28:35.598028789 +0100
@@ -115,8 +115,10 @@
 		head0 = sizeof(struct psf_header);
 
 		head = head0 + fontsize * unit;
+		/*
 		if (head > len || (!hastable && head != len))
 			bb_error_msg_and_die("input file: bad length");
+		*/
 	} else {
 		/* file with three code pages? */
 		if (len == 9780) {
