diff -ur busybox-0.60.1/libbb/xfuncs.c busybox-0.60.1.new/libbb/xfuncs.c
--- busybox-0.60.1/libbb/xfuncs.c	Fri Mar 16 23:47:14 2001
+++ busybox-0.60.1.new/libbb/xfuncs.c	Sat Aug 18 11:40:15 2001
@@ -35,7 +35,12 @@
 #ifndef DMALLOC
 extern void *xmalloc(size_t size)
 {
-	void *ptr = malloc(size);
+	void *ptr;
+
+	if (size == 0)
+		size++;
+		
+	ptr = malloc(size);
 
 	if (!ptr)
 		error_msg_and_die(memory_exhausted);
