--- busybox-1.29.3/include/libbb.h~	2018-12-16 11:20:56.000000000 +0100
+++ busybox-1.29.3/include/libbb.h	2018-12-16 11:22:32.895687694 +0100
@@ -2095,6 +2095,7 @@ extern const char bb_default_login_shell
 # define VC_3 "/dev/tty3"
 # define VC_4 "/dev/tty4"
 # define VC_5 "/dev/tty5"
+# define VC_12 "/dev/tty12"
 # define VC_FORMAT "/dev/tty%d"
 #elif ENABLE_FEATURE_DEVFS
 /*Linux, obsolete devfs names */
--- busybox-1.29.3/init/init.c~	2018-07-02 13:23:06.000000000 +0200
+++ busybox-1.29.3/init/init.c	2018-12-16 11:21:50.307764617 +0100
@@ -221,7 +221,7 @@ struct globals {
 #define G (*(struct globals*)bb_common_bufsiz1)
 #define INIT_G() do { \
 	setup_common_bufsiz(); \
-	IF_NOT_FEATURE_INIT_SYSLOG(G.log_console = VC_5;) \
+	IF_NOT_FEATURE_INIT_SYSLOG(G.log_console = VC_12;) \
 } while (0)
 
 enum {
