diff -urN busybox-1.00-pre2.org/modutils/insmod.c busybox-1.00-pre2/modutils/insmod.c
--- busybox-1.00-pre2.org/modutils/insmod.c	2003-08-02 23:46:49.000000000 +0200
+++ busybox-1.00-pre2/modutils/insmod.c	2003-08-05 23:34:53.000000000 +0200
@@ -1890,6 +1890,9 @@
 		struct obj_symbol *sym;
 		char *name = (char *)s->name;
 
+		if (strncmp(name, "GPLONLY_", 8) == 0)
+			name += 8;
+
 #ifdef SYMBOL_PREFIX
 		/* Prepend SYMBOL_PREFIX to the symbol's name (the
 		   kernel exports `C names', but module object files

