--- Config-initrd.h	2003-07-01 03:06:16.000000000 +0200
+++ Config-initrd.h	2003-07-01 03:10:00.000000000 +0200
@@ -9,68 +9,68 @@
 // BusyBox Applications
 //#define BB_ADJTIMEX
 //#define BB_AR
-#define BB_ASH
-#define BB_BASENAME
+//#define BB_ASH
+//#define BB_BASENAME
 #define BB_CAT
-#define BB_CHGRP
-#define BB_CHMOD
-#define BB_CHOWN
-#define BB_CHROOT
-#define BB_CHVT
-#define BB_CLEAR
+//#define BB_CHGRP
+//#define BB_CHMOD
+//#define BB_CHOWN
+//#define BB_CHROOT
+//#define BB_CHVT
+//#define BB_CLEAR
 //#define BB_CMP
-#define BB_CP
+//#define BB_CP
 //#define BB_CPIO
-#define BB_CUT
-#define BB_DATE
+//#define BB_CUT
+//#define BB_DATE
 //#define BB_DC
-#define BB_DD
+//#define BB_DD
 //#define BB_DEALLOCVT
-#define BB_DF
-#define BB_DIRNAME
+//#define BB_DF
+//#define BB_DIRNAME
 #define BB_DMESG
 //#define BB_DOS2UNIX
 //#define BB_DPKG
 //#define BB_DPKG_DEB
 //#define BB_DUTMP
-#define BB_DU
+//#define BB_DU
 //#define BB_DUMPKMAP
 #define BB_ECHO
 #define BB_ENV
 //#define BB_EXPR
 //#define BB_FBSET
 //#define BB_FDFLUSH
-#define BB_FIND
-#define BB_FREE
-//#define BB_FREERAMDISK
+//#define BB_FIND
+//#define BB_FREE
+#define BB_FREERAMDISK
 //#define BB_FSCK_MINIX
 //#define BB_GETOPT
-#define BB_GREP
-#define BB_GUNZIP
-#define BB_GZIP
-#define BB_HALT
-#define BB_HEAD
+//#define BB_GREP
+//#define BB_GUNZIP
+//#define BB_GZIP
+//#define BB_HALT
+//#define BB_HEAD
 //#define BB_HOSTID
 //#define BB_HOSTNAME
-//#define BB_HUSH
-#define BB_ID
-//#define BB_IFCONFIG
-#define BB_INIT
-//#define BB_INSMOD
+#define BB_HUSH
+//#define BB_ID
+#define BB_IFCONFIG
+//#define BB_INIT
+#define BB_INSMOD
 #define BB_INSMOD_NG
 #define BB_KILL
-#define BB_KILLALL
-#define BB_KLOGD
+//#define BB_KILLALL
+//#define BB_KLOGD
 //#define BB_LASH
 //#define BB_LENGTH
-#define BB_LN
+//#define BB_LN
 //#define BB_LOADACM
 //#define BB_LOADFONT
 //#define BB_LOADKMAP
-#define BB_LOGGER
+//#define BB_LOGGER
 //#define BB_LOGNAME
 //#define BB_LOSETUP
-#define BB_LS
+//#define BB_LS
 #define BB_LSMOD
 //#define BB_MAKEDEVS
 //#define BB_MD5SUM
@@ -78,72 +78,72 @@
 //#define BB_MKFIFO
 //#define BB_MKFS_MINIX
 #define BB_MKNOD
-#define BB_MKSWAP
+//#define BB_MKSWAP
 //#define BB_MKTEMP
-#define BB_MODPROBE
-#define BB_MORE
+//#define BB_MODPROBE
+//#define BB_MORE
 #define BB_MOUNT
 //#define BB_MSH
 //#define BB_MT
-#define BB_MV
+//#define BB_MV
 //#define BB_NC
 //#define BB_NSLOOKUP
-#define BB_PIDOF
+//#define BB_PIDOF
 //#define BB_PING
-//#define BB_PIVOT_ROOT
-#define BB_POWEROFF
+#define BB_PIVOT_ROOT
+//#define BB_POWEROFF
 //#define BB_PRINTF
-#define BB_PS
+//#define BB_PS
 #define BB_PWD
 #define BB_RAID_START
 //#define BB_RDATE
 //#define BB_READLINK
-#define BB_REBOOT
+//#define BB_REBOOT
 //#define BB_RENICE
-#define BB_RESET
+//#define BB_RESET
 #define BB_RM
 #define BB_RMDIR
 //#define BB_RMMOD
-//#define BB_ROUTE
+#define BB_ROUTE
 //#define BB_RPM2CPIO
-#define BB_SED
+//#define BB_SED
 //#define BB_SETKEYCODES
 #define BB_SLEEP
-#define BB_SORT
+//#define BB_SORT
 //#define BB_STTY
-#define BB_SWAPONOFF
-#define BB_SYNC
-#define BB_SYSLOGD
-#define BB_TAIL
-#define BB_TAR
+//#define BB_SWAPONOFF
+//#define BB_SYNC
+//#define BB_SYSLOGD
+//#define BB_TAIL
+//#define BB_TAR
 //#define BB_TEE
 //#define BB_TEST
 //#define BB_TELNET
 //#define BB_TFTP
 //#define BB_TIME
 //#define BB_TOP
-#define BB_TOUCH
+//#define BB_TOUCH
 //#define BB_TR
 //#define BB_TRACEROUTE
-#define BB_TRUE_FALSE
-#define BB_TTY
+//#define BB_TRUE_FALSE
+//#define BB_TTY
 //#define BB_UNIX2DOS
 //#define BB_UUENCODE
 //#define BB_UUDECODE
 #define BB_UMOUNT
-#define BB_UNIQ
-#define BB_UNAME
+//#define BB_UNIQ
+//#define BB_UNAME
 //#define BB_UPDATE
 #define BB_UPTIME
-//#define BB_USLEEP
+#define BB_USLEEP
 //#define BB_VI
 //#define BB_WATCHDOG
-#define BB_WC
+//#define BB_WC
 //#define BB_WGET
-#define BB_WHICH
-#define BB_WHOAMI
-#define BB_XARGS
-#define BB_YES
+//#define BB_WHICH
+//#define BB_WHOAMI
+//#define BB_XARGS
+//#define BB_YES
 // End of Applications List
 //
 //
@@ -156,8 +156,8 @@
 //
 // If you enabled one or more of the shells, you may select which one
 // should be run when sh is invoked:
-#define BB_FEATURE_SH_IS_ASH
-//#define BB_FEATURE_SH_IS_HUSH
+//#define BB_FEATURE_SH_IS_ASH
+#define BB_FEATURE_SH_IS_HUSH
 //#define BB_FEATURE_SH_IS_LASH
 //#define BB_FEATURE_SH_IS_MSH
 //
@@ -184,7 +184,7 @@
 //#define BB_FEATURE_USE_DEVPS_PATCH
 //
 // show verbose usage messages
-#define BB_FEATURE_VERBOSE_USAGE
+//#define BB_FEATURE_VERBOSE_USAGE
 //
 // Use termios to manipulate the screen ('more' is prettier with this on)
 //#define BB_FEATURE_USE_TERMIOS
@@ -250,7 +250,7 @@
 //
 // If you are using uClibc, be sure that you've already compiled
 // uClibc with INCLUDE_RPC=true (contained in the Config file)
-//#define BB_FEATURE_NFSMOUNT
+#define BB_FEATURE_NFSMOUNT
 //
 // Enable support forced filesystem unmounting 
 // (i.e., in case of an unreachable NFS system).
@@ -273,11 +273,11 @@
 //
 // Enable command line editing in the shell.  
 // Only relevant if a shell is enabled. On by default.
-#define BB_FEATURE_COMMAND_EDITING
+//#define BB_FEATURE_COMMAND_EDITING
 //
 // Enable tab completion in the shell.  This is now working quite nicely.
 // This feature adds a bit over 4k. Only relevant if a shell is enabled.
-#define BB_FEATURE_COMMAND_TAB_COMPLETION
+//#define BB_FEATURE_COMMAND_TAB_COMPLETION
 //
 // Attempts to match usernames in a ~-prefixed path
 //#define BB_FEATURE_COMMAND_USERNAME_COMPLETION
@@ -286,7 +286,7 @@
 //were shell builtins.  Nice for staticly linking an emergency rescue shell,
 //among other things. Off by default.
 // Only relevant if a shell is enabled.
-//#define BB_FEATURE_SH_STANDALONE_SHELL
+#define BB_FEATURE_SH_STANDALONE_SHELL
 //
 //When this is enabled, busybox shell applets can be called using full path
 //names.  This causes applets (i.e., most busybox commands) to override
