diff -ur busybox-1.35.0.orig/include/libbb.h busybox-1.35.0/include/libbb.h
--- busybox-1.35.0.orig/include/libbb.h	2021-12-26 16:53:26.000000000 +0000
+++ busybox-1.35.0/include/libbb.h	2022-04-03 22:02:20.060835065 +0000
@@ -312,11 +312,19 @@
  * gcc will throw warnings on printf("%d", off_t)... Have to use %ld etc.
  */
 # if UINT_MAX == ULONG_MAX
+#ifdef __ILP32__
+typedef unsigned long long uoff_t;
+#  define XATOOFF(a) xatoull_range((a), 0, LLONG_MA_STRTOOFF bb_strX)
+#  define BBtoull
+#  define STRTOOFF strtoull
+#  define OFF_FMT "ll"
+#else
 typedef unsigned long uoff_t;
 #  define XATOOFF(a) xatoi_positive(a)
 #  define BB_STRTOOFF bb_strtou
 #  define STRTOOFF strtol
 #  define OFF_FMT "l"
+#endif
 # else
 typedef unsigned long uoff_t;
 #  define XATOOFF(a) xatoul_range((a), 0, LONG_MAX)
